File: 백준/Silver/4673. 셀프 넘버/셀프 넘버.c
#include <stdio.h>
int self_num(int n){
    int sum=n;
    while(n>0){
        sum+=n%10;
        n=n/10;
    }
    if(sum>10000)
        sum=0;
    else
        sum;
    return sum;
}

int main(){
    int arr[10001]={0,};
    for (int i=0;i<10001;i++){
        arr[self_num(i)]=1;
    }
    for (int j=0;j<10001;j++){
        if(arr[j]!=1)
            printf("%d\n",j);
    }
}

File: 백준/Silver/4673. 셀프 넘버/셀프 넘버.swift
func selfNum(_ n: Int) -> Int {
    var sum = n
    var num = n
    while num > 0 {
        sum += num % 10
        num /= 10
    }
    if sum > 10000 { sum = 0 }
    return sum
}

var arr = [Int](repeating: 0, count: 10001)
var z = ""
for i in 0..<10001 { arr[selfNum(i)] = 1 }
for j in 0..<10001 { if arr[j] != 1 { z += "\(j)\n" } }
print(z)